### PR TITLE
Document the JDBC SQL connector

### PR DESCRIPTION
**Pull request by JakeSCahill**
Relies on #514

### CODE REVIEW

File: docs/modules/sql/pages/mapping-to-jdbc.adoc [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+{description}
+
+== What is the JDBC Connector
+
+The JDBC connector allows you to connect to any database that supports the JDBC interface.
+
+== Installing the Connector
+
+This connector is included in the full and slim distributions of Hazelcast.
```
Comment (angelasimms) [inline docs/modules/sql/pages/mapping-to-jdbc.adoc:12]:
```suggestion
The JDBC connector is included in the full and slim distributions of Hazelcast.
```

File: docs/modules/sql/pages/mapping-to-jdbc.adoc [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+[.enterprise]*Enterprise*
+
+If xref:security:enabling-jaas.adoc[security] is enabled, your clients may need permissions to use this connector. For details, see xref:pipelines:job-security.adoc[].
+
+== Before you Begin
+
+Before you can create a mapping to an external data store, you must have the following:
+
+- A JDBC driver that's compatible with your data store. This driver must be on your cluster members' classpaths:
+** The full distribution comes with a JDBC driver for MySQL and PostgresSQL data stores.
```
Comment (angelasimms) [inline docs/modules/sql/pages/mapping-to-jdbc.adoc:24]:
```suggestion
** The full distribution of Hazelcast comes with a JDBC driver for MySQL and PostgreSQL data stores.
```

File: docs/modules/sql/pages/mapping-to-jdbc.adoc [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+** The slim distribution does not come with a JDBC driver.
```
Comment (angelasimms) [inline docs/modules/sql/pages/mapping-to-jdbc.adoc:25]:
```suggestion
** The slim distribution of Hazelcast does not come with a JDBC driver.
```

File: docs/modules/sql/pages/mapping-to-jdbc.adoc [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      jdbcUrl: jdbc:mysql://dummy:3306
+      username: xyz
+      password: xyz
+    shared: true
+```
+
+You then create the following `people` table:
+
+```sql
+CREATE TABLE (id INT PRIMARY KEY, name VARCHAR(100))
```
Comment (angelasimms) [inline docs/modules/sql/pages/mapping-to-jdbc.adoc:48]:
```suggestion
CREATE TABLE people (id INT PRIMARY KEY, name VARCHAR(100))
```

**Review by angelasimms — APPROVED** [review]
All looks good to me.

File: docs/modules/sql/pages/mapping-to-jdbc.adoc [hunk 5]
```diff
@@ -0,0 +1,67 @@
+= Mapping to JDBC
+:description: To query external data stores, you need to create a mapping with the JDBC connector so that the SQL service knows how to access the data store.
```
Comment (JakeSCahill) [inline docs/modules/sql/pages/mapping-to-jdbc.adoc:2]:
```suggestion
:description: To query external data stores, you can create a mapping to them with the JDBC connector.
```

File: docs/modules/sql/pages/mapping-to-jdbc.adoc [hunk 6]
```diff
@@ -0,0 +1,67 @@
+= Mapping to JDBC
+:description: To query external data stores, you can create a mapping to them with the JDBC connector.
```
Comment (JakeSCahill) [inline docs/modules/sql/pages/mapping-to-jdbc.adoc:2]:
```suggestion
:description: To query external data stores, you can create a mapping to them with the JDBC connector.
:page-beta: true
```

**Review by frant-hartm — CHANGES_REQUESTED** [review]

File: docs/modules/sql/pages/mapping-to-jdbc.adoc [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+== Before you Begin
+
+Before you can create a mapping to an external data store, you must have the following:
+
+- A JDBC driver that's compatible with your data store. This driver must be on the classpath of your cluster members:
+** The full distribution of Hazelcast comes with a JDBC driver for MySQL and PostgreSQL data stores.
+** The slim distribution of Hazelcast does not come with a JDBC driver.
+- A xref:external-data-stores:external-data-stores.adoc[data store connection] that's configured on all cluster members.
+
+== Creating a JDBC Mapping
```
[the remark this answers is not in the document]
Comment (JakeSCahill) [reply docs/modules/sql/pages/mapping-to-jdbc.adoc:29]: We don't have a section about using the mapping or supported statements, but if there is a limited list of statements that you can use, we should definitely document those here. Do we have a list?

Also, are there any details that users need to be able to use the mapping?

File: docs/modules/sql/pages/mapping-to-jdbc.adoc [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+  mysql-database: <3>
+    class-name: com.hazelcast.datastore.JdbcDataStoreFactory
+    properties:
+      jdbcUrl: jdbc:mysql://dummy:3306
+      username: xyz
+      password: xyz
+    shared: true
+```
+
+You then create the following `people` table:
```
Comment (JakeSCahill) [inline docs/modules/sql/pages/mapping-to-jdbc.adoc:46]:
```suggestion
You then create the following `people` table in the MySQL database:
```

File: docs/modules/sql/pages/mapping-to-jdbc.adoc [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+```sql
+CREATE MAPPING people (
+  <1>
+  id INT, 
+  name VARCHAR 
+) 
+TYPE JDBC <2>
+OPTIONS ( 
+  'externalDataStoreRef'='mysql-database' <3>
+)
```
[the remark this answers is not in the document]
Comment (JakeSCahill) [reply docs/modules/sql/pages/mapping-to-jdbc.adoc:63]: are there other options that are prefixed with `jdbc`?

[the remark this answers is not in the document]
Comment (frant-hartm) [reply docs/modules/sql/pages/mapping-to-jdbc.adoc:63]: At the moment there isn't.

File: docs/modules/sql/pages/mapping-to-jdbc.adoc (hunk 7 above)
[the remark this answers is not in the document]
Comment (frant-hartm) [reply docs/modules/sql/pages/mapping-to-jdbc.adoc:29]: In 5.2 we support `SELECT`, `INSERT`, `UPDATE` and `DELETE`.

**Review by frant-hartm — APPROVED** [review]

**Pull request merged**